### DISCUSSION TITLE
Fail to create "Filebeat" with release package 4.2.0

### DISCUSSION THREAD

**Discussion by lyr3051** [Axway-API-Management-Plus/apigateway-openlogging-elk#193] [Q&A] [answered]
Hi

I tried to use the release package 4.2.0 to create the "filebeat" container, but it return the error message.
<img width="1440" alt="image" src="https://user-images.githubusercontent.com/43023893/159215940-2a8bf961-cb56-4c13-8290-347ff9d3a765.png">

I tried the below method and adjusted the setting of volumes in docker-compose.filebeat.yml, the filebeat container is created successfully.
https://stackoverflow.com/questions/59904878/docker-compose-volume-name-is-too-short-names-should-be-at-least-two-alphanume
<img width="1401" alt="image" src="https://user-images.githubusercontent.com/43023893/159216701-eb242bd5-7418-47fb-b6cb-b0295cbc6cf6.png">

Have you met this issue in your lab before?
Is this adjustment the correct way to fix the issue in this solution?

**Comment by cwiechmann** [comment, marked as answer]
That's very likely because docker-compose cannot find the `.env` file and that's why many of required variables are missing. Have you used the following command? 

`docker-compose --env-file .env -f filebeat/docker-compose.filebeat.yml up -d`

**Reply by lyr3051** [reply]
Hi @cwiechmann 
I used the old command  `docker-compose -f filebeat/docker-compose.filebeat.yml`.

However, if I don't adjust the setting of the volumes, it still responses with the same error message after trying the command you provided.

**Reply by cwiechmann** [reply]
Still, I saw this error, when the attributes (e.g. `APIGATEWAY_TRACES_FOLDER`) for the volumes are missing. Then, you end up with the following instruction:
`- :/var/log/trace`
Which is obviously not correct. 

Now, as you have added the addtional `/` you essentially end up with the following instead:
`- /:/var/log/trace`
Now, the container is created, but very likely the volumes are not correctly mapped, because you just map the root of your filesystem into the container instead of the log-directory.

Please check the parameters in your .env file:
```
APIGATEWAY_OPENTRAFFIC_FOLDER=/home/localuser/Axway-x.y.z/apigateway/logs/opentraffic
APIGATEWAY_TRACES_FOLDER=/home/localuser/Axway-x.y.z/apigateway/groups/group-2/instance-1/trace
APIGATEWAY_EVENTS_FOLDER=/home/localuser/Axway-x.y.z/apigateway/events
APIGATEWAY_AUDITLOGS_FOLDER=/home/localuser/Axway-x.y.z/apigateway/logs
APIGATEWAY_PAYLOADS_FOLDER=/home/localuser/Axway-x.y.z/apigateway/logs/payloads
```

And make sure, the .env file is really used. There is no need to change the docker-compose file for filebeat.

**Reply by lyr3051** [reply]
Hi @cwiechmann 
Thanks for your help.
The issue is resolved because I misconfigure a parameter in my .env so that the path of the parameter couldn't be find.
